icu4c/source/i18n/units_complexconverter.cpp

```diff
@@ -132,24 +132,20 @@ MaybeStackVector<Measure> ComplexUnitsConverter::convert(double quantity,
     for (int i = 0, n = unitConverters_.length(); i < n; ++i) {
         quantity = (*unitConverters_[i]).convert(quantity);
         if (i < n - 1) {
-            // The double type has 15 decimal digits of precision. For choosing
-            // whether to use the current unit or the next smaller unit, we
-            // therefore nudge up the number with which the thresholding
-            // decision is made. However after the thresholding, we use the
-            // original values to ensure unbiased accuracy (to the extent of
-            // double's capabilities).
-            int64_t roundedQuantity = floor(quantity * (1 + DBL_EPSILON));
-            intValues[i] = roundedQuantity;
+            // If quantity is at the limits of double's precision from an
+            // integer value, we take that integer value.
+            int64_t flooredQuantity = floor(quantity * (1 + DBL_EPSILON));
```

Comment:
Now that we have support for a `RoundingImpl *rounder` parameter, we could probably drop this DBL_EPSILON nudge by setting a default `rounder` instead. I'm happy to add it to this PR if you prefer, otherwise I can do this in a later PR.